blocks/api/factory.js

```diff
@@ -67,6 +68,48 @@ export function createBlock( name, blockAttributes = {} ) {
 }
 
 /**
+ * Returns an array of possible block transformations that could happen on the set of blocks received as argument.
+ *
+ * @param  {Array}  blocks Blocks array
+ * @return {Array}         Array of possible block transformations
+ */
+export function getPossibleBlockTransformations( blocks ) {
+	if ( ! blocks || ! blocks[ 0 ] ) {
+		return [];
+	}
+	const isMultiBlock = blocks.length > 1;
+	const sourceBlockName = blocks[ 0 ].name;
+
+	if ( isMultiBlock && ! every( blocks, ( block ) => ( block.name === sourceBlockName ) ) ) {
```

Comment:
Minor: Most Lodash methods allow predicates to be defined using an object shorthand. In this case:

```js
// Before
! every( blocks, ( block ) => ( block.name === sourceBlockName ) )

// After
! every( blocks, { name: sourceBlockName } )
```

The only potential worry is that it is listed as being removed by default in the 5.0.0 roadmap:

https://github.com/lodash/lodash/wiki/Roadmap